### PR TITLE
Add Code Guidelines

### PR DESCRIPTION
**Pull request by ElishaSamPeterPrabhu**
### :page_facing_up: Summary of Changes
- Adding the guidelines to write and review code
### :thought_balloon: Type of Change

- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)
- [ ] Configuration change (modifies scaling or properties of existing infrastructure & services)
- [ ] Documentation change
- [x] Other

### :clipboard: Test Plan

[comment]: # 'How do you know the changes are safe to ship to production?'
[comment]: # 'How do you know the changes will function as expected in future releases?'

### :white_check_mark: Self Code Review Checklist

PR authors and reviewers, please verify that all of these items have been completed.

- [x] My code is clean and readable
- [x] I followed standard conventions
- [ ] Component code is WCAG 2.2 compliant (if applicable)
- [ ] I have made theme files changes (if applicable)
- [ ] I have made documentation changes (if applicable)
- [ ] I have made Storybook changes including usage documentation (if applicable)
- [ ] I have tested the component thoroughly in Storybook including RTL rendering (if applicable)

### :link: Work Item

Issue #

### CODE REVIEW

File: modus-code-guide - review.md [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+- Each component should have its own directory.
+- Directory should contain:
+  - Component file (e.g., `modus-wc-select.tsx`).
+  - Styles file (e.g., `modus-wc-select.scss`).
+  - Test file (e.g., `modus-wc-select.spec.ts`).
+  - Storybook file (e.g., `modus-wc-select.stories.ts`).
+  - README file (e.g., `README.md`).
+
+### Folder Structure
```
Comment (gabriel-piltzer-trimble) [inline modus-code-guide - review.md]: We found that organizing components atomically caused confusion, so we removed atomic folders. Developers typically know which component they're modifying before opening the project.

File: Code Guidelines.md [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+
+### Code Guidelines
+
+#### Code Structure and Organization
+
+1. **Component Structure**:
+
+   - Each component should be self-contained with its own HTML, CSS, and JS/TS files.
+   - Use meaningful and descriptive names for components and their files.
+   - Organize components in a hierarchical folder structure that reflects their relationships and dependencies.
```
Comment (gabriel-piltzer-trimble) [inline Code Guidelines.md]: Not sure if this is true, we just put components and related files in their own folder (except for icons, which are shared from `src/icons`).

File: Code Guidelines.md [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+
+2. **File Naming Conventions**:
+
+   - Use kebab-case for file names (e.g., `modus-wc-text-input.ts`).
+   - Ensure file names are descriptive of their content and purpose.
+
+3. **Component Registration**:
+   - Register custom elements using the `customElements.define` method.
```
Comment (gabriel-piltzer-trimble) [inline Code Guidelines.md]: This is not manually done by the developer

File: Code Guidelines.md [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+   - Ensure the tag name follows the pattern `modus-wc-*` for consistency.
+
+#### Coding Standards
+
+1. **TypeScript Usage**:
+
+   - Prefer TypeScript over JavaScript for type safety and better tooling support.
+   - Define types and interfaces for component properties and events.
```
Comment (gabriel-piltzer-trimble) [inline Code Guidelines.md]: Include component type but not full names in interfaces, so...
`ISelectOption` ✅ 
`IModusWcSelectOption` 🚫

File: Code Guidelines.md [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+
+2. **Property Definitions**:
+
+   - Use the `@Prop` decorator to define component properties.
+   - Provide default values for properties where applicable.
+   - Use appropriate types for properties (e.g., `string`, `boolean`, `number`).
+   - Use `!` for defining important in props. (e.g., `@Prop() alt!: string`)
```
Comment (gabriel-piltzer-trimble) [inline Code Guidelines.md]:
```suggestion
   - Use `!` for defining required props. (e.g., `@Prop() alt!: string`)

```

And `?` for optional props

File: Code Guidelines.md [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+
+3. **Event Emission**:
+
+   - Use the `@Event` decorator to define custom events.
+   - Name events using camelCase (e.g., `inputBlur`, `inputChange`).
+   - Emit events using the `this.emit` method.
+
+4. **CSS Styling**:
+
+   - Use scoped CSS to avoid conflicts with other components.
```
Comment (gabriel-piltzer-trimble) [inline Code Guidelines.md]: We don't use scoped CSS on purpose. Instead we rely on strict class naming conventions, starting with `modus-wc-`. This may change a little with the Daisy 5 upgrade. Could move to `modus:` but not 100% sure yet.

File: Code Guidelines.md [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+   - Use CSS variables for styling (eg. `--modus-wc-color-green-dark`). If a CSS value is being repeated, consider creating a new CSS variable or set of them(eg. `--modus-wc-font-size-xs`).
+   - Follow a consistent naming convention for CSS classes (e.g., `modus-wc-text-input__label`).
```
Comment (gabriel-piltzer-trimble) [inline Code Guidelines.md]: We don't necessarily use `__` in our naming. But some components are using CSS BEM naming.

File: Code Guidelines.md [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+   - Use DaisyUI classes whenever possible via tailwind typescripts (e.g, `modus-wc-avatar.tailwind.ts`).
```
Comment (gabriel-piltzer-trimble) [inline Code Guidelines.md]:
```suggestion
   - Dynamically added DaisyUI and Tailwind classes are added in their own files (e.g, `modus-wc-avatar.tailwind.ts`).

```

File: Code Guidelines.md [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+5. **Consistent Code Style**:
+
+   - Follow a consistent code style throughout the codebase. Use tools like Prettier and ESLint to enforce code style rules.
+   - Use single quotes for strings, and ensure proper indentation and spacing.
+   - Add blank lines between styling blocks for readability and consistency.
+
+6. **Error Handling**:
+
+   - Implement proper error handling in components and functions. Use try-catch blocks where necessary and provide meaningful error messages.
+   - Ensure that errors are logged appropriately for debugging purposes.
```
Comment (gabriel-piltzer-trimble) [inline Code Guidelines.md]: But also don't log frivolously as this will pollute the console of various Trimble applications.

File: Code Guidelines.md [hunk 10]
```diff
@@ -0,0 +1,310 @@
...
+   - Use `inheritedAttributes` to set aria properties which is available in the utils file as if type `Attributes`, and call `inheritAriaAttributes()` to set it.
+
+- Use [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web) as reference for best practices.
+
+#### Version Control and Collaboration
+
+1. **Commit Messages**:
+
+   - Write clear and concise commit messages.
+   - Use the imperative mood in commit messages (e.g., "Add {new-feature}", "Fix {bug}").
```
Comment (gabriel-piltzer-trimble) [inline Code Guidelines.md]: I like semantic commit messages like "feature: Add {new-feature}" and "fix: {bug}"

File: Code Guidelines.md [hunk 11]
```diff
@@ -0,0 +1,310 @@
...
+   - Do not commit unused imports.
+
+2. **Pull Requests**:
+
+   - Provide a clear description of the changes and the rationale behind them for each and every PR.
+   - Clarify in detail on the review comments promptly and update the pull request accordingly with the fix.
```
Comment (gabriel-piltzer-trimble) [inline Code Guidelines.md]: Add an item telling them to fill out the entire PR template

File: Code Guidelines.md [hunk 12]
```diff
@@ -0,0 +1,310 @@
...
+
+   - Use descriptive names for test cases.
+   - Follow the pattern `should do something when condition` for test names.
+
+3. **Code Coverage**:
+
+   - Aim for 100% code coverage, but prioritize meaningful tests over coverage percentage.
+   - Update test cases to cover the code changes for each component.
+
+4. **End-to-End Testing**:
```
Comment (gabriel-piltzer-trimble) [inline Code Guidelines.md]: We don't really have e2e tests currently.

File: Code Guidelines.md [hunk 13]
```diff
@@ -0,0 +1,310 @@
...
+
+   - Write end-to-end tests for critical user flows to ensure the respective component or feature works as expected.
+
+5. **Continuous Integration**:
+   - Ensure continuous integration (CI) pipelines which run tests and checks on each commit and pull request to be succesful.
+
+#### Documentation and Comments
+
+1. **Component Documentation**:
```
Comment (gabriel-piltzer-trimble) [inline Code Guidelines.md]: Add a line that we use the auto docs feature in Storybook to automatically populate some of the storybook information.

File: Code Guidelines.md [hunk 14]
```diff
@@ -0,0 +1,310 @@
...
+   - Provide migration docs if applicable after each Pull requests
+
+2. **Inline Comments**:
+
+   - Use inline comments to explain complex logic and important decisions.
+   - Keep comments concise and relevant.
+
+3. **README Files**:
+
+   - Each component should have a `readme.md` file that provides an overview of the component, its properties, events, and usage examples.
```
Comment (gabriel-piltzer-trimble) [inline Code Guidelines.md]:
```suggestion
   - Each component should have an auto-generated `readme.md` file that provides an overview of the component, its properties, events, and usage examples.

```

File: Code Guidelines.md [hunk 15]
```diff
@@ -0,0 +1,310 @@
...
+
+### Example Code Snippet
+
+````typescript
+import { Component, Prop, Event, EventEmitter, h } from '@stencil/core';
+
+@Component({
+  tag: 'modus-wc-text-input',
+  styleUrl: 'modus-wc-text-input.css',
+  shadow: true,
```
Comment (gabriel-piltzer-trimble) [inline Code Guidelines.md]: We never want this to be true

```suggestion
  shadow: false,

```

File: Code Guidelines.md [hunk 16]
```diff
@@ -0,0 +1,310 @@
...
+5. **Configuration Files**:
+   - Keep configuration files (e.g., `tsconfig.json`) at the root level of the project for easy access and management.
+   - Ensure configuration files are well-documented with comments explaining the purpose of each configuration option.
+
+### Example Code Snippet
+
+````typescript
+import { Component, Prop, Event, EventEmitter, h } from '@stencil/core';
+
+@Component({
```
Comment (gabriel-piltzer-trimble) [inline Code Guidelines.md]: This example component lacks all of the proper JS Doc comments that we use.

File: Code Guidelines.md [hunk 17]
```diff
@@ -0,0 +1,310 @@
...
+  tag: 'modus-wc-text-input',
+  styleUrl: 'modus-wc-text-input.css',
+  shadow: true,
+})
+export class ModusWcTextInput {
+  @Prop() label: string;
+  @Prop() placeholder: string;
+  @Prop() value: string;
+  @Prop() disabled: boolean = false;
```
Comment (gabriel-piltzer-trimble) [inline Code Guidelines.md]: Alphabetize props and add single line comments above each.

File: Code Guidelines.md [hunk 18]
```diff
@@ -0,0 +1,310 @@
...
+
+   - Ensure documentation is present for any events exposed which is generated by autoDocs in the storybook.
+   - Ensure documentation includes examples and descriptions of each component generated by autoDocs in the storybook.
+   - AutoDocs generates information with the the comments of each `event` or `@Prop`.
+
+5. **Configuration Files**:
+   - Keep configuration files (e.g., `tsconfig.json`) at the root level of the project for easy access and management.
+   - Ensure configuration files are well-documented with comments explaining the purpose of each configuration option.
+
+### Example Code Snippet
```
Comment (gabriel-piltzer-trimble) [inline Code Guidelines.md]: I think that maintaining this provides less value than pointing them at one of our smaller/simpler components actual code.

File: Code Guidelines.md [hunk 19]
```diff
@@ -0,0 +1,310 @@
...
+          disabled={this.disabled}
+          onInput={(event) => this.handleInputChange(event)}
+        />
+      </div>
+    );
+  }
+}
+````
+
+## Contribution Guidelines
```
Comment (gabriel-piltzer-trimble) [inline Code Guidelines.md]: Can you review this @coliff ?

**Comment by coliff** [comment]
A few comments:

- I think all of this could be added to `CONTRIBUTING.md`
- If you do decide you'd prefer for this to be separate then there should be links to it from `CONTRIBUTING.md` and `README.md`
- The filename should not have spaces and for consistency with other markdown files should be all-caps... so `CODE-GUIDE.md`
- The file should have a h1 heading at the very top (line 1) e.g. `# Code Guide`

File: modus-code-guide - review.md (hunk 1 above)
Comment (Kavin-Trimble) [reply modus-code-guide - review.md]: Addressed

File: Code Guidelines.md (hunk 2 above)
Comment (Kavin-Trimble) [reply Code Guidelines.md]: Addressed

File: Code Guidelines.md (hunk 3 above)
Comment (Kavin-Trimble) [reply Code Guidelines.md]: Addressed

File: Code Guidelines.md (hunk 4 above)
Comment (Kavin-Trimble) [reply Code Guidelines.md]: Addressed

File: Code Guidelines.md (hunk 5 above)
Comment (Kavin-Trimble) [reply Code Guidelines.md]: Addressed

File: Code Guidelines.md (hunk 6 above)
Comment (Kavin-Trimble) [reply Code Guidelines.md]: Okay, removed the scoped CSS line. For now the naming convention consists of `modus-wc`. In the future based on what we decide, we can change it to `modus`

File: Code Guidelines.md (hunk 7 above)
Comment (Kavin-Trimble) [reply Code Guidelines.md]: Changed the example (e.g., `modus-wc-text-input-label`). I couldn't find any components using CSS BEM. Do we need to add the convention here? @gabriel-piltzer-trimble

File: Code Guidelines.md (hunk 8 above)
Comment (Kavin-Trimble) [reply Code Guidelines.md]: Addressed

File: Code Guidelines.md (hunk 9 above)
Comment (Kavin-Trimble) [reply Code Guidelines.md]: Addressed

File: Code Guidelines.md (hunk 10 above)
Comment (Kavin-Trimble) [reply Code Guidelines.md]: Addressed

File: Code Guidelines.md (hunk 11 above)
Comment (Kavin-Trimble) [reply Code Guidelines.md]: Addressed

File: Code Guidelines.md (hunk 12 above)
Comment (Kavin-Trimble) [reply Code Guidelines.md]: Removed e2e tests

File: Code Guidelines.md (hunk 13 above)
Comment (Kavin-Trimble) [reply Code Guidelines.md]: Added

File: Code Guidelines.md (hunk 14 above)
Comment (Kavin-Trimble) [reply Code Guidelines.md]: Addressed

File: Code Guidelines.md (hunk 15 above)
Comment (Kavin-Trimble) [reply Code Guidelines.md]: Addressed

File: Code Guidelines.md (hunk 16 above)
Comment (Kavin-Trimble) [reply Code Guidelines.md]: Addressed this by linking it to [ModusWcButton](https://github.com/trimble-oss/modus-wc-2.0/blob/main/src/components/modus-wc-button/modus-wc-button.tsx)

File: Code Guidelines.md (hunk 17 above)
Comment (Kavin-Trimble) [reply Code Guidelines.md]: Addressed this by linking it to [ModusWcButton](https://github.com/trimble-oss/modus-wc-2.0/blob/main/src/components/modus-wc-button/modus-wc-button.tsx)

File: Code Guidelines.md (hunk 18 above)
Comment (Kavin-Trimble) [reply Code Guidelines.md]: I agree; hence addressed this by linking it to [ModusWcButton](https://github.com/trimble-oss/modus-wc-2.0/blob/main/src/components/modus-wc-button/modus-wc-button.tsx)

**Comment by Kavin-Trimble** [comment]
> A few comments:
> 
> * I think all of this could be added to `CONTRIBUTING.md`
> * If you do decide you'd prefer for this to be separate then there should be links to it from `CONTRIBUTING.md` and `README.md`
> * The filename should not have spaces and for consistency with other markdown files should be all-caps... so `CODE-GUIDE.md`
> * The file should have a h1 heading at the very top (line 1) e.g.CODE-GUIDE.md`

@coliff 

- Since the file `CODE-GUIDE.md` is used for both reviewing and writing code, can it be separate and linked in the `CONTRIBUTING.md` and `README.md` files?
- File name is changed to `CODE-GUIDE.md` and `# Code Guide` is added to the top of the file.

**Comment by coliff** [comment]
> Since the file CODE-GUIDE.md is used for both reviewing and writing code, can it be separate and linked in the CONTRIBUTING.md and README.md files?

Yep, ok. thanks for renaming :-)

File: CODE-GUIDE.md [hunk 20]
```diff
@@ -0,0 +1,256 @@
...
+   - Define types and interfaces for component properties and events.
+   - Add component type on naming interfaces but full names are not needed to be included. (e.g., `ISelectOption` (not as `IModusWcSelectOption`))
+
+2. **Property Definitions**:
+
+   - Use the `@Prop` decorator to define component properties.
+   - Provide default values for properties where applicable.
+   - Use appropriate types for properties (e.g., `string`, `boolean`, `number`).
+   - Use `!` for defining required props. (e.g., `@Prop() alt!: string`)
+   - Use `?` for defining required props. (e.g., `@Prop() alt?: string`)
```
Comment (gabriel-piltzer-trimble) [inline CODE-GUIDE.md]:
```suggestion
   - Use `?` for defining optional props. (e.g., `@Prop() alt?: string`)
```

File: Code Guidelines.md (hunk 6 above)
Comment (gabriel-piltzer-trimble) [reply Code Guidelines.md]: As long as what you change it to doesn't interfere with the older Modus 1.0 classes. A major requirement is that 1.0 and 2.0 components can render side by side with no negative side effects.

**Review by gabriel-piltzer-trimble — COMMENTED** [review]
One comment, also curious how developers are going to find/discover this file if it's not linked from the `README` or `CONTRIBUTING`.

**Comment by Kavin-Trimble** [comment]
> One comment, also curious how developers are going to find/discover this file if it's not linked from the `README` or `CONTRIBUTING`.

We'll link the CODE-GUIDE.md to CONTRIBUTING.md

**Review by gabriel-piltzer-trimble — DISMISSED** [review]
You have a few spellcheck failures that look legitimate and should be fixed.

File: CODE-GUIDE.md [hunk 21]
```diff
@@ -0,0 +1,256 @@
...
+
+   - Use collaboration tools like GitHub Issues and Project Boards to track tasks, bugs, and feature requests.
+   - Ensure that all team members are aware of the current status of the project and any ongoing work.
+
+#### Testing and Quality Assurance
+
+1. **Unit Testing**:
+
+   - Write unit tests for all components and functions.
+   - We are using jest testing framework consistently throughout the project ensure tests cover various scenarios, including edge cases.
```
Comment (Kai-Richardson) [inline CODE-GUIDE.md]:
```suggestion
   - We are using the Jest testing framework throughout the project to ensure tests cover various scenarios, including edge cases.
```
copyediting

File: CONTRIBUTING.md [hunk 22]
```diff
@@ -1,5 +1,6 @@
 # Contributing
 
+- Adhere to the Guidelines [CODE-GUIDE.md](https://github.com/trimble-oss/modus-wc-2.0/blob/main/CODE-GUIDE.md) while contributing.
```
Comment (Kai-Richardson) [inline CONTRIBUTING.md]: Are these guidelines or a guide? The wording is conflicting here and in the actual document.

Those two words tend to have different meanings, at least in my experience.
* 'Guide' tends to mean a tutorial on how to do something, more optional.
* 'Guidelines' tends to mean something that is required, a set of conditions you can check.

File: CODE-GUIDE.md [hunk 23]
```diff
@@ -0,0 +1,256 @@
...
+- Ensure your code follows the coding standards.
+- Write tests for your changes.
+- Submit a pull request with a clear description of your changes.
+
+# Need Discussion
+
+1. **Branching Strategy**:
+
+   - Create branches using the following format
+     - IssueNo-IssueTitle (e.g, 109-collapse-fix-animation)
```
Comment (Kai-Richardson) [inline CODE-GUIDE.md]: What if I'm making a PR and there's so associated issue?

File: CODE-GUIDE.md [hunk 24]
```diff
@@ -0,0 +1,256 @@
...
+
+2. **End-to-End Testing**: - Use a consistent testing framework - playwright to be explored and followed for `.e2e` tests.
```
Comment (Kai-Richardson) [inline CODE-GUIDE.md]: Do we use playwright at all right now? Should we be mentioning this if we're not using it yet?

File: CODE-GUIDE.md [hunk 25]
```diff
@@ -0,0 +1,256 @@
...
+#### Documentation and Comments
+
+- NOTE: The storybook creates its information using the auto documentation feature; in order to produce accurate information, inline comments must be provided.
+
+1. **Component Documentation**:
+
+   - Provide storybook inline docs for all components
+   - Provide component details in the `stories.ts` file to generate inline docs for components.
+   - Include types, and default values in the component properties.
+   - Add `ifDefined` wherever needed, but not for Boolean values.
```
Comment (Kai-Richardson) [inline CODE-GUIDE.md]: What is 'wherever needed'? Would it be easier for people if we clarified it was for optional values?

File: CODE-GUIDE.md [hunk 26]
```diff
@@ -0,0 +1,256 @@
...
+
+   - Write clear and concise commit messages.
+   - Use the semantic mood in commit messages (e.g., "feature: Add {new-feature}" and "Fix {bug}").
+   - Do not commit unused imports.
+
+2. **Pull Requests**:
+
+   - Provide a clear description of the changes and the rationale behind them for each and every PR.
+   - Clarify in detail on the review comments promptly and update the pull request accordingly with the fix.
+   - Fill all the items in the PR template on creating PRs
```
Comment (Kai-Richardson) [inline CODE-GUIDE.md]:
```suggestion
   - Fill in all the items in the PR template when creating PRs.
```

File: CODE-GUIDE.md [hunk 27]
```diff
@@ -0,0 +1,256 @@
...
+   - Use `inheritedAttributes` to set aria properties which is available in the utils file as if type `Attributes`, and call `inheritAriaAttributes()` to set it.
+
+- Use [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web) as reference for best practices.
+
+#### Version Control and Collaboration
+
+1. **Commit Messages**:
+
+   - Write clear and concise commit messages.
+   - Use the semantic mood in commit messages (e.g., "feature: Add {new-feature}" and "Fix {bug}").
```
Comment (Kai-Richardson) [inline CODE-GUIDE.md]: What is semantic mood? I have not heard of this before.

File: CODE-GUIDE.md [hunk 28]
```diff
@@ -0,0 +1,256 @@
...
+
+   - Implement proper error handling in components and functions. Use try-catch blocks where necessary and provide meaningful error messages.
+   - Ensure that errors are logged appropriately for debugging purposes (but avoid frivolous logging).
+
+7. **Accesibility**
+
+   - Ensure all components are accessible and follow WCAG 2.2 standards.
+   - Components should be fully operable via keyboard alone.
+   - Use ARIA attributes where necessary to enhance accessibility.
+   - Use `inheritedAttributes` to set aria properties which is available in the utils file as if type `Attributes`, and call `inheritAriaAttributes()` to set it.
```
Comment (Kai-Richardson) [inline CODE-GUIDE.md]: The grammar in this sentence is a bit off, I'm not confident that I know what this is trying to convey.
I think we should also clarify (if we're going into such detail) when to call `inheritAriaAttributes()`, since you can't just call it anywhere in the component lifecycle.

File: CODE-GUIDE.md [hunk 29]
```diff
@@ -0,0 +1,256 @@
...
+  - Component file (e.g., `modus-wc-select.tsx`).
+  - Styles file (e.g., `modus-wc-select.scss`).
+  - Test file (e.g., `modus-wc-select.spec.ts`).
+  - Storybook file (e.g., `modus-wc-select.stories.ts`).
+  - README file (e.g., `README.md`).
+
+### Folder Structure
+
+```
+modus-wc-2.0/
```
Comment (Kai-Richardson) [inline CODE-GUIDE.md]: What's the value of having this full structure here with a bunch of assorted files?
I get the thought behind documenting the broad information architecture pattern, but what's the reasoning behind mentioning a bunch of loose files that contributors should never need to touch?
e.g., tsconfig.json or the theme provider

File: CODE-GUIDE.md [hunk 30]
```diff
@@ -0,0 +1,256 @@
...
+
+### Component Organization
+
+- Each component should have its own directory.
+- Directory should contain:
+  - Component file (e.g., `modus-wc-select.tsx`).
+  - Styles file (e.g., `modus-wc-select.scss`).
+  - Test file (e.g., `modus-wc-select.spec.ts`).
+  - Storybook file (e.g., `modus-wc-select.stories.ts`).
+  - README file (e.g., `README.md`).
```
Comment (Kai-Richardson) [inline CODE-GUIDE.md]: perhaps mention this (the readme) is automatically generated

**Review by Kai-Richardson — CHANGES_REQUESTED** [review]
Assorted copyediting and structural review, feel free to reply to the comments if you have any questions.

If I comment on a code section, I am commenting on the line directly above my comment, not the whole auto-included block.

File: CODE-GUIDE.md (hunk 24 above)
Comment (Kavin-Trimble) [reply CODE-GUIDE.md]: We intend to use the Playwright Framework to automate the e2e tests. I'll remove this point for the time being.

File: CODE-GUIDE.md (hunk 27 above)
Comment (Kavin-Trimble) [reply CODE-GUIDE.md]: It is a convention for writing commit messages in a standardized format that provides clear and meaningful information about the changes being made. 

I got this suggestion from Gabe's [review comment](https://github.com/trimble-oss/modus-wc-2.0/pull/364#discussion_r2060515418).

File: CODE-GUIDE.md (hunk 30 above)
Comment (Kavin-Trimble) [reply CODE-GUIDE.md]: Addressed

File: CODE-GUIDE.md (hunk 28 above)
Comment (Kavin-Trimble) [reply CODE-GUIDE.md]: Addressed

File: CODE-GUIDE.md (hunk 26 above)
Comment (Kavin-Trimble) [reply CODE-GUIDE.md]: Addressed

File: CODE-GUIDE.md (hunk 25 above)
Comment (Kavin-Trimble) [reply CODE-GUIDE.md]: Addressed

File: CONTRIBUTING.md (hunk 22 above)
Comment (Kavin-Trimble) [reply CONTRIBUTING.md]: Updated the filename as CODE-GUIDELINES

File: CODE-GUIDE.md (hunk 27 above)
Comment (Kai-Richardson) [reply CODE-GUIDE.md]: Those are called semantic commit messages (or [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)), not semantic mood.

File: CODE-GUIDE.md (hunk 21 above)
Comment (Kavin-Trimble) [reply CODE-GUIDE.md]: Addressed

File: CODE-GUIDE.md (hunk 23 above)
Comment (Kavin-Trimble) [reply CODE-GUIDE.md]: If you meant "no associated issue," I've included a format for that as well.

File: CODE-GUIDE.md (hunk 27 above)
Comment (Kavin-Trimble) [reply CODE-GUIDE.md]: OK, I've updated this.

File: CODE-GUIDE.md (hunk 29 above)
Comment (Kavin-Trimble) [reply CODE-GUIDE.md]: Updated

File: CONTRIBUTING.md (hunk 22 above)
Comment (Kai-Richardson) [reply CONTRIBUTING.md]: this link is broken, then

File: CODE-GUIDELINES.md [hunk 31]
```diff
@@ -0,0 +1,225 @@
...
+   - Use descriptive names for test cases.
+   - Follow the pattern `should do something when condition` for test names.
+
+3. **Code Coverage**:
+
+   - Aim for 100% code coverage, but prioritize meaningful tests over coverage percentage.
+   - Update test cases to cover the code changes for each component.
+
+4. **Continuous Integration**:
+   - Ensure continuous integration (CI) pipelines which run tests and checks on each commit and pull request to be succesful.
```
Comment (Kai-Richardson) [inline CODE-GUIDELINES.md]: spelling
```suggestion
   - Ensure continuous integration (CI) pipelines which run tests and checks on each commit and pull request to be successful.
```

File: CODE-GUIDELINES.md [hunk 32]
```diff
@@ -0,0 +1,225 @@
...
+   - Follow a consistent code style throughout the codebase. Use tools like Prettier and ESLint to enforce code style rules.
+   - Use single quotes for strings, and ensure proper indentation and spacing.
+   - Add blank lines between styling blocks for readability and consistency.
+
+6. **Error Handling**:
+
+   - Implement proper error handling in components and functions. Use try-catch blocks where necessary and provide meaningful error messages.
+   - Ensure that errors are logged appropriately for debugging purposes (but avoid frivolous logging).
+
+7. **Accesibility**
```
Comment (Kai-Richardson) [inline CODE-GUIDELINES.md]:
```suggestion
7. **Accessibility**
```

**Review by Kai-Richardson — COMMENTED** [review]
looks a lot better 👍, just the 4 minor changes

File: CODE-GUIDELINES.md [hunk 33]
```diff
@@ -0,0 +1,225 @@
...
+- Write tests for your changes.
+- Submit a pull request with a clear description of your changes.
+
+# Need Discussion
+
+1. **Branching Strategy**:
+
+   - Create branches using the following format
+     - IssueNo-IssueTitle (e.g, 109-collapse-fix-animation)
+     - ComponentName-IssueDescription {feature/bug} (e.g, Table-add-summaryRow)
```
Comment (Kai-Richardson) [inline CODE-GUIDELINES.md]: Is this intended to be a main header category of needs discussion? Should this be included if we haven't decided on it yet?

Comment (Kavin-Trimble) [reply CODE-GUIDELINES.md]: No, I added that remark earlier when I needed our teammates here to clarify some points. We had talked about it, and I've removed it now.

File: CODE-GUIDELINES.md (hunk 32 above)
Comment (Kavin-Trimble) [reply CODE-GUIDELINES.md]: Updated

File: CODE-GUIDELINES.md (hunk 31 above)
Comment (Kavin-Trimble) [reply CODE-GUIDELINES.md]: Updated

File: CONTRIBUTING.md (hunk 22 above)
Comment (Kavin-Trimble) [reply CONTRIBUTING.md]: The link has been updated, and it will be accessible in Main after this PR is merged. Right now, this local branch can be used to access it.

File: CONTRIBUTING.md [hunk 34]
```diff
@@ -1,5 +1,7 @@
 # Contributing
 
+- Adhere to the Guidelines [CODE-GUIDE.md](https://github.com/trimble-oss/modus-wc-2.0/blob/main/CODE-GUIDELINES.md) while contributing.
```
Comment (Kai-Richardson) [inline CONTRIBUTING.md:3]:
```suggestion
- Adhere to the [Code Guidelines](https://github.com/trimble-oss/modus-wc-2.0/blob/main/CODE-GUIDELINES.md) while contributing.
```

**Review by Kai-Richardson — COMMENTED** [review]
Looks great otherwise.

**Review by coliff — DISMISSED** [review]

**Review by Kai-Richardson — CHANGES_REQUESTED** [review]
linters

**Review by coliff — APPROVED** [review]

**Review by Kai-Richardson — APPROVED** [review]
